src/Client.js

```diff
@@ -0,0 +1,92 @@
+// @flow
+import { Axios } from 'axios';
+import Versions from './api/versions';
+import Certifications from './api/certifications/Certifications';
+import type { $AxiosXHRConfigBase, $AxiosXHR, $AxiosXHRConfig, $AxiosError } from 'axios';
+
+class AxiosError extends Error {
+    response: $AxiosXHR<*>;
+    config: $AxiosXHRConfig<*>;
+}
+const DEFAULT_OPTS = {
+    baseURL: 'https://api.app.gigwalk.com'
+};
+export default class Gigwalk extends Axios {
+    constructor(config?: $AxiosXHRConfigBase<*> = DEFAULT_OPTS) {
+        super(config);
+        this.versions = new Versions(this);
+        this.certifications = new Certifications(this);
+        this.queue = new Map();
+    }
+
+    versions: Versions;
+    certifications: Certifications;
+    defaults: {
+        headers: {
+            common: {
+                Authorization?: string
+            }
+        }
+    };
+    queue: Map<string, Promise<$AxiosXHR<*>>>;
+
+    setAuthToken(token: string): Promise<$AxiosXHR<*>> {
+        const prefix: string = 'Token ';
+        const Authorization = token.indexOf(prefix) === 0 ? token : `${prefix}${token}`;
+        return this.get('/v1/auth', {
+            headers: { Authorization }
+        }).then((res: $AxiosXHR<*>): $AxiosXHR<*> => {
+            this.defaults.headers.common.Authorization = Authorization;
+            return res;
+        });
+    }
+
+    request(config: $AxiosXHRConfig<any>, ...args: Array<any>): Promise<$AxiosXHR<*>> {
```

Comment:
I am just extending the base Axios class which gives me a lot of power. I am going to sue the `Map` object and use a similar approach you did, but the API is a bit more simple since it is basically just removing the queue if the request was already taken care of.